bullet/src/EntityManagementFeatures.cc

```diff
@@ -29,6 +29,108 @@ Identity EntityManagementFeatures::ConstructEmptyWorld(
       {world, _name, collisionConfiguration, dispatcher, broadphase, solver});
 }
 
+/////////////////////////////////////////////////
+bool EntityManagementFeatures::RemoveModel(const Identity &_modelID)
+{
+  // Check if the model exists
+  if (this->models.find(_modelID) == this->models.end()){
+    return false;
+  }
+
+  const auto &modelInfo = this->models.at(_modelID);
+
+  // Current implementation does not include collisions nor joints
+  // Those should be removed here before removing the model
+
+  // Clean up links
+  for (const auto &linkEntry : this->links)
+  {
+    const auto &linkInfo = linkEntry.second;
+    if (linkInfo->model.id == _modelID.id)
+    {
+      this->links.erase(linkEntry.first);
+    }
+  }
+
+  // (To-DO blast545): test if its required to make an extra call to
+  // world->removeMultiBody(model) to remove it from the bullet system
+  // or is it enough just by deleting its pointed memory
+
+  // Clean up model
+  delete modelInfo->model;
+  this->models.erase(_modelID);
+
+  return true;
+}
+
+bool EntityManagementFeatures::ModelRemoved(
+  const Identity &_modelID) const
+{
+  return this->models.find(_modelID) == this->models.end();
+}
+
+bool EntityManagementFeatures::RemoveModelByIndex(
+  const Identity &/* _worldID */, std::size_t _modelIndex)
```

Comment:
Identity of the model and Index of the model seem to be the same right? Cant you just use `RemoveModel(_modelIndex)` ?

Reply:
`RemoveModelByIndex` receives a `size_t` index, while the other version receives an Identity object, I could rewrite the one receiving the identity to call this one instead, but I'm not sure in which contexts these functions are used. So we can reconsider this if needed later after we test the features